near-sdk/src/json_types/public_key.rs

```diff
@@ -97,31 +128,38 @@ impl<'de> serde::Deserialize<'de> for Base58PublicKey {
     where
         D: serde::Deserializer<'de>,
     {
-        let s: String = Deserialize::deserialize(deserializer)?;
+        let s: String = serde::Deserialize::deserialize(deserializer)?;
         s.parse::<Base58PublicKey>().map_err(serde::de::Error::custom)
     }
 }
 
 impl From<&Base58PublicKey> for String {
     fn from(str_public_key: &Base58PublicKey) -> Self {
-        match str_public_key.0[0] {
-            0 => "ed25519:".to_string() + &bs58::encode(&str_public_key.0[1..]).into_string(),
-            1 => "secp256k1:".to_string() + &bs58::encode(&str_public_key.0[1..]).into_string(),
-            _ => panic!("Unexpected curve"),
-        }
+        let upper_bound = (str_public_key.data.len() / 5 + 1) * 8;
```

Comment:
I'll look more into this tomorrow, but where are the values for this coming from? This seems like it might be above the amount needed for base58

Reply:
This should be in `bs58/src/encode.rs::318`. You can additionally try to get to it via going to the definition of `into` within `bs58::encode(&str_public_key.data).into(&mut buf[pad..]).unwrap();`